pkg/codegen/docs/gen.go

```diff
@@ -1757,11 +1757,37 @@ func (mod *modContext) gen(fs codegen.Fs) error {
 	dctx := mod.docGenContext
 	modName := mod.getModuleFileName()
 
-	addFile := func(name, contents string) {
-		p := path.Join(modName, name, "_index.md")
+	seen := map[string]struct{}{}
+
+	addFile := func(name, contents string, prefixes []string) {
+		var namePrefixed string
+		for _, prefix := range prefixes {
+			namePrefixed = prefix + name
+			if _, exists := seen[namePrefixed]; exists {
+				// Unset namePrefixed because it isn't valid.
+				namePrefixed = ""
```

Comment:
Style: Set the variable and then unset it like that is a bit off in terms of flow.
How about:

```go
var found bool
for _, prefix := range prefixes {
  candidate := prefix + name
  if _, exists := seen[candidate]; exists {
    continue
  }
  name = candidate
  found = true
  seen[name] = struct{}
  break
}

if !found {
  return
}
```

The target is set only if the candidate is valid.

(My example doesn't have a namePrefixed because we don't use the original "name" variable again after this point, and the prefixed version effectively replaces it.)

Reply:
> similar names with different casing: GetName and getName.
I _feel like_ this pattern of naming doesn't exist in practice (and shouldn't exist in practice) because it's not idiomatic in any supported language. I would expect if someone created a provider with these names in the schema, we should encourage them to get a little more creative with their names. I would expect we could issue guidance around naming conventions in the schema to mitigate this concern.

Just my 2c, feel free to ignore me :)